crates/db_schema/src/lib.rs

```diff
@@ -175,6 +175,7 @@ pub enum SearchType {
   Communities,
   Users,
   Url,
+  PostTitle,
```

Comment:
No, PostTitle isn't a search return type, its a filter. You can remove this.

Reply:
Seems like I merged this too quickly. Actually the way its implemented now means you can search for communities or comments with `title_only = true` and it wont make any difference. Thats quite unintuitive, if its only for post titles then adding SearchType::PostTitle makes the most sense. And no SearchType is not the result type, thats SearchResponse.

Reply:
Should I switch to the `SearchType::PostTitle` flow again?

Reply:
Yes please make a new pull request with that change.

Reply:
I disagree with this, just make your searchtype Post, and then use titleonly. In the front end we can only show the title only filter for all or post type searches. 

Yes some search filters only apply to certain return types, and there are possibly better ways to do it (bundle specific filters with the return type), but having them separate works fine.

Reply:
I dont see any other search parameter which only works with a single search type. Doing it like this is very unintuitive for client developers. Also we are talking about the SearchType, not the return type (which is SearchResponse).

Reply:
Check out which fields the [Person search uses](https://github.com/LemmyNet/lemmy/blob/main/crates/apub/src/api/search.rs#L95). Doesn't use the creator or community filters.

Either way its best to separate filters from return types. The SearchType determines what type of info is returned in that list (whethers its posts only, comments only, persons only, etc), so its functionally the return type.